In the same `test-26.3` environment, I get: 
```
>>> import setuptools_scm
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    import setuptools_scm
ModuleNotFoundError: No module named 'setuptools_scm'
```
So that probably explains the lack of any version string

--- a/py/desiutil/install.py
+++ b/py/desiutil/install.py
@@ -965,6 +965,11 @@ class DesiInstall(object):
         can be used to generate a version string, which can then be written to
         a version file.
 
+        In the event that this particular use of ``setup.py`` is finally deprecated,
+        ``setuptools-scm`` can and should be moved to the ``pyproject.toml`` file.
+
+        When installing a *tagged* version, this function does nothing.
+
         For tagged installs, :meth:`~desiutil.install.DesiInstall.prepare_environment`
         sets :envvar:`SETUPTOOLS_SCM_PRETEND_VERSION_FOR_PACKAGE` with ``PACKAGE``
         replaced with the package name. This allows :command:`pip install` to

--- a/py/desiutil/install.py
+++ b/py/desiutil/install.py
@@ -976,6 +976,7 @@ except Exception:
     version = '{version}'
 """.lstrip()
                 version_command = [sys.executable, '-m', 'setuptools_scm']
+            self.log.debug(' '.join(version_command))
             proc = Popen(version_command, universal_newlines=True,
                          stdout=PIPE, stderr=PIPE)
             out, err = proc.communicate()

--- a/py/desiutil/install.py
+++ b/py/desiutil/install.py
@@ -955,6 +955,21 @@ class DesiInstall(object):
     def compile_version(self):
         """Generate a version string for main/branch installs of packages
         that set the version string dynamically, *i.e.* with ``setuptools-scm``.
+
+        As of Spring 2026, only two DESI packages, ``speclite`` and ``specsim``,
+        use this method, and the configuration for ``setuptools-scm`` is stored
+        their ``setup.py`` files. :command:`python setup.py --version` will
+        create the necessary version file.
+
+        However, if there is no ``setup.py`` file, :command:`python -m setuptools_scm`
+        can be used to generate a version string, which can then be written to
+        a version file.
+
+        For tagged installs, :meth:`~desiutil.install.DesiInstall.prepare_environment`
+        sets :envvar:`SETUPTOOLS_SCM_PRETEND_VERSION_FOR_PACKAGE` with ``PACKAGE``
+        replaced with the package name. This allows :command:`pip install` to
+        set the version, even though the GitHub tarballs do not contain
+        git metadata such as the last tag or commit ID.
         """
         if self.is_branch and self.baseproduct in setuptools_scm_products:
             current_dir = os.getcwd()

